### PR TITLE
add basic to_one_hot function

### PR DESCRIPTION
**Pull request by justheuristic**
Hello!
This may be covered in some pending pull-request, but if it isn't, here's a simple one-hot function from my collection of pytorch helpers that may or may not be useful :)

It essentially copies [this issue](https://discuss.pytorch.org/t/convert-int-into-one-hot-format/507).

### CODE REVIEW

**Comment by soumith** [comment]
@justheuristic can you add a test to test_nn.py (see other tests for functional, but can be something simple: give known input and check for known result).

Also, add an entry to: https://github.com/pytorch/pytorch/blob/master/docs/source/nn.rst

**Comment by justheuristic** [comment]
Sorry for the delay. Added simple test case and autodoc.

**Comment by justheuristic** [comment]
Sorry for the delay. Added simple test case and autodoc.

**Comment by justheuristic** [comment]
@soumith
Is there anything else to add?

**Comment by justheuristic** [comment]
Almighty @pytorchbot , help me again!

**Comment by ezyang** [comment]
Test failures are legit

**Comment by justheuristic** [comment]
i wonder if i can do that

File: torch/nn/functional.py [hunk 1]
```diff
@@ -2027,6 +2027,51 @@ def assert_int_or_pair(arg, arg_name, message):
...
+        (1 ,.,.) =
+          0  0  1
+          1  0  0
+
+        (2 ,.,.) =
+          0  1  0
+          0  0  1
+        [torch.FloatTensor of size 3x2x3]
+    """
+    y_tensor = y.data if isinstance(y, Variable) else y
```
Comment (weiyangfb) [inline torch/nn/functional.py]: This needs to migrate to 0.4.0+

**Comment by zou3519** [comment]
Sorry for missing this @justheuristic. We merged a one_hot implementation in #15208; thank you for your PR nonetheless

**Pull request closed without merging**